client/packages/system/src/IndicatorsDemographics/DetailView/IndicatorsDemographics.tsx

```diff
@@ -83,21 +63,24 @@ const IndicatorsDemographicsComponent: FC = () => {
   } = useUrlQueryParams({
     initialSort: { key: 'percentage', dir: 'desc' },
   });
+  const [headerDraft, setHeaderDraft] = useState<HeaderData>();
   const [indexPopulation, setIndexPopulation] = useState<number>();
-
-  const { draft, setDraft } =
-    useDemographicData.document.listIndicator(headerData);
-
-  const draftHeaders = ArrayUtils.toObject(headerData);
   const [isDirty, setIsDirty] = useState(false);
 
-  const [headerDraft, setHeaderDraft] =
-    useState<Record<string, HeaderValue>>(draftHeaders);
+  const { error, success } = useNotification();
+  const t = useTranslation();
+
+  const { draft, setDraft } = useDemographicData.indicator.list(headerDraft);
+  const { data: projections, isLoading: isLoadingProjections } =
+    useDemographicData.projection.list(draft?.[0]?.baseYear ?? 2024);
 
   const { insertDemographicIndicator, invalidateQueries } =
-    useDemographicData.document.insertIndicator();
+    useDemographicData.indicator.insert();
   const { mutateAsync: updateDemographicIndicator } =
-    useDemographicData.document.updateIndicator();
+    useDemographicData.indicator.update();
+  const upsertProjection = useDemographicData.projection.upsert();
+  const getHeaderDraft = () =>
```

Comment:
probs not required now that the hook is returning something. but can't hurt I guess.

Reply:
This is just adding indirection/confusion for me?